lib/matplotlib/tests/test_cycles.py

```diff
@@ -104,6 +126,45 @@ def test_fillcycle_ignore():
     ax.legend(loc='upper left')
 
 
+@cleanup
+def test_valid_input_forms():
+    fig, ax = plt.subplots()
+    # These should not raise an error.
+    ax.set_prop_cycle(None)
+    ax.set_prop_cycle(cycler('linewidth', [1, 2]))
+    ax.set_prop_cycle('color', 'rgywkbcm')
+    ax.set_prop_cycle('linewidth', (1, 2))
+    ax.set_prop_cycle('linewidth', [1, 2])
+    ax.set_prop_cycle('linewidth', iter([1, 2]))
+    ax.set_prop_cycle('linewidth', np.array([1, 2]))
+    ax.set_prop_cycle('color', np.array([[1, 0, 0],
+                                         [0, 1, 0],
+                                         [0, 0, 1]]))
+    ax.set_prop_cycle(lw=[1, 2], color=['k', 'w'], ls=['-', '--'])
+    ax.set_prop_cycle(lw=np.array([1, 2]),
+                      color=np.array(['k', 'w']),
+                      ls=np.array(['-', '--']))
+    assert True
+
+
+@cleanup
+def test_invalid_input_forms():
+    fig, ax = plt.subplots()
+    with assert_raises((TypeError, ValueError)):
```

Comment:
can't use assert_raises as a context manager if you want this to go into the 1.5.x branch. Nose doesn't provide that context manager for python 2.6 for some reason.